XLS-47d-PriceOracles/README.md

```diff
@@ -0,0 +1,468 @@
+<pre> 
+   Title:        <b>Price Oracles on XRP Ledger</b>
+   Revision:     <b>3</b> (2023-10-18)
+ <hr>  Author: <a href="mailto:gtsipenyuk@ripple.com">Gregory Tsipenyuk</a>
+   Affiliation:  <a href="https://ripple.com">Ripple</a>
+ 
+ </pre>
+ 
+ # Price Oracles on XRP Ledger
+ 
+ ## Abstract
+
+ This proposal adds an on-chain `PriceOracle` object to the XRPL ledger. A blockchain oracle is a system or service that acts as a bridge between a blockchain network and the external world, providing off-chain data or information to decentralized applications (dApps) on the blockchain. Oracles are used to bring real-world data, for instance  market prices, exchange rates, interest rates, or weather conditions onto the blockchain, enabling dApps to access and utilize information that resides outside the blockchain. This document outlines a new protocol for price oracles the on XRPL ledger, and provides guidelines for developers and system architects to implement and utilize this solution effectively. This proposal introduces a new on-ledger `PriceOracle` object and the transactions to create, delete, and update the `PriceOracle`. It also adds the `get_aggregate_price` API, to retrieve an `aggregate mean`, `trimmed mean`, and `median` for the provided price oracles. This feature requires an amendment.
+
+### Terminology
+
+- **Oracle Provider**: A service or technology that enables the integration of external data and real-world events into a blockchain network.
+- **dApp (Decentralized Application)**: An application that is built on a blockchain network and operates using smart contracts or other mechanisms or protocols for their functionality.
+
+## Creating `PriceOracle` instance on XRPL
+
+### On-Ledger Data Structures
+
+#### The `PriceOracle` Object
+
+The `PriceOracle` ledger entry represents the `PriceOracle` object on XRPL ledger and contains the following fields:
+
+|FieldName | Required? | JSON Type | Internal Type|
+|:---------|:-----------|:---------------|:---------------|
+| `LedgerEntryType`| :heavy_check_mark: | `string` | `UINT16` |
+| `Owner` | :heavy_check_mark: | `string` | `ACCOUNTID` |
+| `Provider` | :heavy_check_mark: | `string` | `BLOB` |
+| `PriceDataSeries` | :heavy_check_mark: | `array` | `ARRAY` |
+| `LastUpdateTime` | :heavy_check_mark: | `number` | `UINT32` |
```

Comment:
how about using string as ISO8601-compliant time, with second precision ? I heard that some users have problems using ripple-epoch time.

Reply:
FWIW I do not feel strongly about ISO8601, but it would lower adoption bar a little.

Reply:
WE already use ripple epoch time on-ledger in objects like escrows, I'd rather be internally consistent.